src/lib.rs

```diff
@@ -290,17 +324,44 @@ impl Certificate {
 						});
 						writer.next().write_bytes(&bytes);
 					});
+					if let IsCa::Ca(ref constraint) = self.params.is_ca {
+						// Write subject_key_identifier
+						writer.next().write_sequence(|writer| {
+							let oid = ObjectIdentifier::from_slice(OID_SUBJECT_KEY_IDENTIFIER);
+							writer.next().write_oid(&oid);
+							let digest = digest::digest(&digest::SHA256, self.key_pair.public_key().as_ref());
+							writer.next().write_bytes(&digest.as_ref());
+						});
+						// Write basic_constraints
+						writer.next().write_sequence(|writer| {
+							let oid = ObjectIdentifier::from_slice(OID_BASIC_CONSTRAINTS);
+							writer.next().write_oid(&oid);
+							let bytes = yasna::construct_der(|writer| {
+								writer.write_sequence(|writer| {
+									writer.next().write_bool(true);
```

Comment:
Maybe add a terse comment about what this bool means? Like `// cA flag` or something.

Reply:
Done.